SteamKit2/SteamKit2/Networking/Steam3/UdpConnection.cs

```diff
@@ -356,22 +366,49 @@ private bool DispatchMessage()
         /// <summary>
         /// Processes incoming packets, maintains connection consistency, and oversees outgoing packets.
         /// </summary>
-        private void NetLoop()
+        private void NetLoop(object param)
         {
             // Variables that will be used deeper in the function; locating them here avoids recreating
             // them since they don't need to be.
-            EndPoint packetSender = (EndPoint) new IPEndPoint(IPAddress.Any, 0);
+            var userRequestedDisconnect = false;
+            EndPoint packetSender = (EndPoint)new IPEndPoint(IPAddress.Any, 0);
             byte[] buf = new byte[2048];
 
-            timeOut = DateTime.Now.AddSeconds(TIMEOUT_DELAY);
-            nextResend = DateTime.Now.AddSeconds(RESEND_DELAY);
+            if (param is IPEndPoint)
+            {
+                remoteEndPoint = param as IPEndPoint;
+            }
+            else
+            {
+                var epTask = param as Task<IPEndPoint>;
+                try
+                {
+                    remoteEndPoint = epTask.Result;
+                }
+                catch (Exception ex)
+                {
+                    DebugLog.WriteLine("UdpConnection", "Unable to find endpoint to connect, endpoint task returned: {0}", ex);

```

Comment:
FYI, `ex` won't be useful here because it will either be a `NullReferenceException` as above, or an `AggregateException`.